includes/marketplace-suggestions/class-wc-marketplace-suggestions-api.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * REST API Marketplace suggestions API
+ *
+ * Handles requests for marketplace suggestions data
+ *
+ * @package WooCommerce/API
+ * @since   3.6.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Pull suggestion data from remote endpoint & cache in a transient.
+ *
+ * @return array of json API data
+ */
+function wc_marketplace_suggestions_get_api_data() {
+	$suggestion_data_url = 'https://d3t0oesq8995hv.cloudfront.net/add-ons/marketplace-suggestions.json';
+
+	$suggestion_data = get_transient( 'wc_marketplace_suggestions' );
+
+	if ( false !== $suggestion_data ) {
+		return $suggestion_data;
+	}
+
+	$raw_suggestions = wp_safe_remote_get(
+		$suggestion_data_url,
+		array( 'user-agent' => 'WooCommerce Marketplace Suggestions' )
+	);
+	if ( is_wp_error( $raw_suggestions ) ) {
+		return array();
+	}
+
+	// Parse the data to check for any errors.
+	// If it's valid, store structure in transient.
+	$suggestions = json_decode( wp_remote_retrieve_body( $raw_suggestions ) );
```

Comment:
Does this needs some kind of sanitization callback on the contents? The JSON is untrusted data.

Reply:
Good question – I'm relying on `json_decode` to bail if the data is malformed, and if that succeeds, there's a basic check that it's an array.

Regarding the actual contents, I could add checking for certain keys – I'm not sure how useful that is. Happy to take advice as to the best approach here.